pipeline-notebooks/pipeline-general.ipynb

```diff
@@ -540,19 +551,44 @@
    "outputs": [],
    "source": [
     "# pipeline-api\n",
-    "def pipeline_api(file, filename='', response_type=\"application/json\"):\n",
+    "\n",
+    "class UnsupportedFileError(Exception):\n",
+    "    def __init__(self, message):\n",
+    "        self.message = message\n",
+    "        super().__init__(self.message)\n",
+    "\n",
+    "DEFAULT_MIMETYPES = \"application/pdf,application/msword,image/jpeg,image/png,text/markdown,\" \\\n",
+    "                    \"text/x-markdown,application/epub,application/epub+zip,text/html,\" \\\n",
+    "                    \"application/vnd.openxmlformats-officedocument.spreadsheetml.sheet,\" \\\n",
+    "                    \"application/vnd.ms-excel,application/vnd.openxmlformats-officedocument.\" \\\n",
+    "                    \"presentationml.presentation,\" \\\n",
+    "                    \"application/vnd.ms-powerpoint,application/xml\"\n",
+    "\n",
+    "\n",
+    "if not os.environ.get(\"UNSTRUCTURED_ALLOWED_MIMETYPES\", None):\n",
+    "    os.environ[\"UNSTRUCTURED_ALLOWED_MIMETYPES\"] =  DEFAULT_MIMETYPES\n",
+    "\n",
+    "\n",
+    "def pipeline_api(file, filename='', file_content_type=None, file_filename=\"\", response_type=\"application/json\"):\n",
```

Comment:
this signature should not have `file_filename`.

This pipeline_api is primarily intended for API's, in which case `filename` means the name of the `file`, even though it later gets passed along as `file_filename` to partition. `pipeline_api` explicitly does not by convention support opening a local file referenced to by a filename (as `partition` does).

I.e., first three args/kwargs should match: https://github.com/Unstructured-IO/unstructured-api-tools/blob/060ead4/test_unstructured_api_tools/pipeline-test-project/prepline_test_project/api/process_file_1.py#L33